util/gen_effect_id.ts

```diff
@@ -0,0 +1,132 @@
+// did different things that are still around.
+//
+// This is a map of id to skill name (for smoke testing/documentation).
+const knownMapping = {
+  'Thundercloud': '164',
+  'Battle Litany': '786',
+  'Right Eye': '1910',
+  'Left Eye': '1454',
+  'Meditative Brotherhood': '1182',
+  'Brotherhood': '1185',
+  'Embolden': '1297',
+  'Technical Finish': '1822',
+  'Sheltron': '1856',
+  'Lord of Crowns': '1876',
+  'Lady of Crowns': '1877',
+  'Divination': '1878',
+  'The Balance': '1882',
+  'The Bole': '1883',
+  'The Arrow': '1884',
+  'The Spear': '1885',
+  'The Ewer': '1886',
+  'The Spire': '1887',
+  'Sword Oath': '1902',
+  'Tactician': '1951',
+  // This is for others, 1821 is for self.
+  'Standard Finish': '2105',
+  'The Wanderer\'s Minuet': '2216',
+  'Mage\'s Ballad': '2217',
+  'Army\'s Paeon': '2218',
+  'Stormbite': '1201',
+  'Caustic Bite': '1200',
+  'Windbite': '129',
+  'Venomous Bite': '124',
+  'Flourishing Fan Dance': '1820',
+  'Higanbana': '1228',
+  'Wildfire': '861',
+  'Chain Stratagem': '1221',
+  'Vulnerability Up': '638',
+} as const;
+
+// These custom name of effect will not be checked, but you'd better make it clean.
+// Use this only when you need to handle different effects with a same name.
+const customMapping = {
+  'EmboldenSelf': '1239',
+} as const;
+
+
+const printError = (
+    header: string,
+    what: string,
+    map: Record<string | number, unknown>,
+    key: string,
+) =>
+  console.error(`${header} ${what}: ${JSON.stringify(map[key])}`);
+
+
+const makeEffectMap = (table: Table<'#', 'Name'>) => {
+  const foundNames = new Set();
+
+  const map: { [s: string]: string } = {};
+  for (const [id, effect] of Object.entries(table)) {
+    const rawName = effect['Name'];
+    if (!rawName)
+      continue;
+    const name = cleanName(rawName);
+
+    if (rawName && rawName in knownMapping) {
+      if (id !== knownMapping[rawName as keyof typeof knownMapping]) {
+        printError('skipping', rawName, table, id);
+        continue;
+      }
+    }
+
+    if (name in map) {
+      printError('collision', name, table, id);
+      printError('collision', name, table, map[name] ?? '');
+      delete map[name];
+      continue;
+    }
+    if (name in foundNames) {
+      printError('collision', name, table, id);
+      continue;
+    }
+
+    foundNames.add(name);
+    map[name] = id;
+  }
+
+  // Make sure everything specified in known_mapping was found in the above loop.
+  for (const entry of Object.entries(knownMapping)) {
+    const name = cleanName(entry[0]);
+    if (name && !(name in foundNames))
+      printError('missing', name, knownMapping, entry[0]);
+  }
+
+  // Add custom effect name for necessary duplicates.
+  for (const [rawName, id] of Object.entries(customMapping))
+    map[rawName] = id;
+
+  // Store ids as hex.
+  return Object.fromEntries(
+      Object.entries(map).map(([key, value]) => [key, parseInt(value).toString(16).toUpperCase()]),
```

Comment:
```suggestion
      Object.entries(map).map(([name, id]) => [name, parseInt(id).toString(16).toUpperCase()]),
```

Suggesting to give these better names.